### DISCUSSION TITLE
Where in Object

### DISCUSSION THREAD

**Discussion by piovezanfernando** [ujjwalguptaofficial/JsStore#280] [General]
Hi, I'm using JsStore and it's meeting my need perfectly. But a question arose after the implementation that I need to ask the question whether or not it is possible to do this type of consultation.

I have a table with the following fields:

```
columns: {
            id: {
                primaryKey: true,
                autoIncrement: false,
                dataType: DATA_TYPE.Number,
            },
            client: {
                dataType: DATA_TYPE.Object
            },
            created_at: {
                dataType: DATA_TYPE.String,
            }
}
```
The insertion I make for this table is as follows
 ```
{
  "client": {
    "id": 10,
    "name": "fernando",
    "number": 1234567
  },
  "created_at": "2022-08-09 00:00:00"
}
```

I understand that the ideal thing is that the client object was a separate table so that it was possible to perform a join and with a direct where in the field of the related table.
But when the base was created this was not implemented and now I need to perform a search directly in the name field of the client object, would this scenario be possible?
Example of how I tried to create the query but in none of the scenarios it worked.

```
connection.select({
            from: this.tableName,
            where: {
                // client.name: 'Fernando'
                // client: '{name: Fernando}'
                // client: '{name: "Fernando"}'
                // client: 'Fernando'
                client: 'name: Fernando'
            }
        })
```
The commented lines are to exemplify what was tested I only used one line at a time to test

**Comment by ujjwalguptaofficial** [comment]
Right now its not possible as indexeddb does not allow multi level indexing as far as i know, but this can be managed by jsstore to do filtering. 

This can be considered a feature.

**Reply by piovezanfernando** [reply]
I understand this limitation, I just needed to confirm, I will then create another table to work with join and execute the query in this new table.

Thank you very much

**Reply by ujjwalguptaofficial** [reply]
@fernandopiovezan1 you can use transaction for your scenario, but you will have to loop through all the data which will be very slow.

https://jsstore.net/tutorial/transaction/

transaction allows you to perform complex scenario, you will be able to select data, filter it return the way you want.

but for better performance, its better if your schema is simplied and their is index on every data you want to filter.

**Reply by piovezanfernando** [reply]
@ujjwalguptaofficial actually the transaction can be an exit, but as you pointed out it may be slow so I think there is nowhere to escape the best way is to redo the tables structure and create direct relationships to perform the search, even if it takes more time to development in the future will facilitate maintenance.
thank you very much for your attention
